providers/base.ts

```diff
@@ -351,10 +351,33 @@ export abstract class ReleaseLookup<Provider extends MetadataProvider, RawReleas
 	}
 }
 
+export interface ApiAccessToken {
+	accessToken: string;
+	validUntilTimestamp: number;
+}
+
 /** Extends `MetadataProvider` with functions common to lookups accessing web APIs. */
 export abstract class MetadataApiProvider extends MetadataProvider {
 	/** Must be implemented to perform a request against the specified URL. */
 	abstract query<Data>(apiUrl: URL, maxTimestamp?: number): Promise<CacheEntry<Data>>;
+
+	/** Returns a cached API access token.
+	 *
+	 * Must be a passed function `requestAccessToken` which will return a promise resolving to a
+	 * `ApiAccessToken`, containing the access token and an expiration Unix timestamp. The result is
+	 * cached and `requestAccessToken` only gets called if the cached token has expired.
+	 */
+	protected async cachedAccessToken(requestAccessToken: () => Promise<ApiAccessToken>): Promise<string> {
+		const cacheKey = `${this.name}:accessKey`;
```

Comment:
Actually I wondered the same while refactoring, but decided against changing it because now we already have tokens stored. But we could change it anyway.

Reply:
Better change it now then, currently there is only one such cache key used in production.

Reply:
I changed the name, it is indeed better in the long run.